zilliqa/src/api/zil.rs

```diff
@@ -103,10 +112,83 @@ fn create_transaction(params: Params, node: &Arc<Mutex<Node>>) -> Result<serde_j
         sig,
     };
 
-    let transaction_hash = node.create_transaction(transaction)?;
-    let transaction_hash = hex::encode(transaction_hash.0);
+    let transaction_hash = node.create_transaction(transaction.clone())?;
+    let transaction_verified: VerifiedTransaction = transaction.verify()?;
 
-    Ok(json!({"TranID": transaction_hash}))
+    match transaction_verified.tx {
+        SignedTransaction::Zilliqa { ref tx, .. } => {
```

Comment:
This smells a bit to me. You create a `TxZilliqa` above, wrap it into a `SignedTransaction`, then unconditionally match on it to get the `TxZilliqa` back again later.

Can we move the `contract_address` calculation to before the call to `node.create_transaction` and then remove any matching on the transaction here?

Reply:
yes, it is smelly - I will rewrite it. The issue is that you need the nonce and the from address, so you neccessarily requre constructing a verified transaction, but it requires a match on that to get the nonce. However, I could just copy the nonce from earlier in the code.